### ISSUE TITLE
NPM gets internal error when running in an LXC (Proxmox)

### ISSUE THREAD

**Issue by NickRO19** [NginxProxyManager/nginx-proxy-manager#579]
After installing NPM in Docker, everything works fine until I try to get a SSL cert.

Privileged container with ports 80 and 443 forwarded to the LXC.
Used the following script to install.
docker run -d \
    --name=nginx-proxy-manager \
    -p 81:8181 \
    -p 80:8080 \
    -p 443:4443 \
    -v /docker/appdata/nginx-proxy-manager:/config:rw \
    jlesage/nginx-proxy-manager

I get an internal error when requesting the SSL cert from Let's Encrypt. It also may just hang sometimes when trying to update a failed request.

Using Ubuntu 18.04 standard template LXC inside Proxmox 6.2-4

**Comment by Oujiii** [comment]
Hey! You didn't mention me so I had no idea you had opened this.

Is there any reason why you are not using jc21's container? I know you said it works on a VM, but I'm just wondering. Also, which db are you using for it?

**Comment by dash74** [comment]
I can confirm that jc21's works in proxmox without any issue. Although I am using it in an unprivileged container. Since docker doesn't usually play well in a privileged container by default.

**Comment by NickRO19** [comment]
> Hey! You didn't mention me so I had no idea you had opened this.
> 
> Is there any reason why you are not using jc21's container? I know you said it works on a VM, but I'm just wondering. Also, which db are you using for it?

Sorry, kinda new to github...

jc21's I have the same exact issue.

**Comment by NickRO19** [comment]
> I can confirm that jc21's works in proxmox without any issue. Although I am using it in an unprivileged container. Since docker doesn't usually play well in a privileged container by default.

Do you mean the other way around? I always use a privileged container, whenever I use unprivileged docker has issues

**Comment by dash74** [comment]
> > I can confirm that jc21's works in proxmox without any issue. Although I am using it in an unprivileged container. Since docker doesn't usually play well in a privileged container by default.
> 
> Do you mean the other way around? I always use a privileged container, whenever I use unprivileged docker has issues

Nope... Here is a screenshot of my lxc container. It's an unprivileged container. 

![image](https://user-images.githubusercontent.com/28696570/91664131-3cea6680-eabb-11ea-9ddf-67ee4cc21bbe.png)

**Comment by NickRO19** [comment]
Interesting... what version of Ubuntu server are you using?

-------- Original Message --------

**Comment by NickRO19** [comment]
I tried the unprivileged container using Ubuntu 18.04.
Still get the same internal error

-------- Original Message --------

**Comment by dash74** [comment]
I'm using Ubuntu 20.04 standard template. As I mentioned, you do have to make sure nesting and keyctl are checked for docker too work. In an unprivileged container. I just tried installing docker again in a privileged container. Both 18.04 and 20.04. I get docker: Error response from daemon: error creating aufs mount. When attempting too run the sudo docker run hello-world. I'm using the repository install method. It appears it has to do with the storage device. driver. Installing docker in an unprivileged container just works for me, without issue.

**Comment by NickRO19** [comment]
Ok, so I do not get that error when installing in a privileged container.
Regardless its working in a unprivileged just fine. I do have those options checked and its running.
As always though, I still get the internal error when requesting a SSL cert from Lets encrypt

-------- Original Message --------

**Comment by dash74** [comment]
Have you looked at the letsencrypt-requests.log? Do any of the logs provide more info?

**Comment by NickRO19** [comment]
I'm pretty new to this so excuse my lack of knowledge.
Where can I find those logs?

-------- Original Message --------

**Comment by dash74** [comment]
There should be a folder /data/logs. It would be created in the directory you're running the command line or yml file. You should see something like this.


![image](https://user-images.githubusercontent.com/28696570/91667294-7e860c00-ead1-11ea-8275-5432fa62b30c.png)

**Comment by NickRO19** [comment]
I am so sorry for delaying in solving my own issue. Family health problems... 
Just to update I have recently switch to trying this on a VM. I always get the same internal error when trying to get a lets encrypt cert.
I have double checked my dns settings, my ports are open, I can reach my admin page and even see the "Congrats" page when going to my website using regular http.
Whenever i try to get a cert still get the internal error.
Also @dash74 I cannot find that folder "data". Where would it be from the "/" directory? Sorry I'm kind of new to this proxy stuff.
